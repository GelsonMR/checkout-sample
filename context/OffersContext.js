import { createContext, useCallback, useContext, useEffect, useState } from 'react'
import { getOffers } from '../client'

export const OffersContext = createContext({})

export function OffersContextProvider({ children }) {
  const [isLoading, setIsLoading] = useState(false)
  const [offers, setOffers] = useState()
  const [selectedOffer, setSelectedOffer] = useState()
  const [error, setError] = useState(false)

  const fetchOffers = useCallback(async () => {
    setIsLoading(true)
    setError(false)
    try {
      const res = await getOffers()
      setOffers(res.data.sort((a, b) => a.order - b.order))
      setSelectedOffer(res.data[0])
    } catch (e) {
      console.error(e)
      setError(true)
    }
    setIsLoading(false)
  }, [])

  useEffect(() => {
    fetchOffers()
  }, [fetchOffers])

  const sharedState = {
    isLoading,
    offers,
    selectedOffer,
    error,
    setSelectedOffer
  }

  return (
    <OffersContext.Provider value={sharedState}>
      {children}
    </OffersContext.Provider>
  )
}

export function useOffers() {
  return useContext(OffersContext)
}
