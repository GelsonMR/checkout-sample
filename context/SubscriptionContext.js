import { createContext, useCallback, useContext, useState } from 'react'
import { subscribe } from '../client'

export const SubscriptionContext = createContext()

export function SubscriptionContextProvider({ children }) {
  const [isLoading, setIsLoading] = useState(false)
  const [subscription, setSubscription] = useState()
  const [error, setError] = useState(false)

  const fetchSubscribe = useCallback(async (params) => {
    setIsLoading(true)
    setError(false)
    try {
      const res = await subscribe(params)
      setSubscription({
        ...res.data,
        ...params
      })
      setIsLoading(false)
      return res
    } catch (e) {
      console.error(e)
      setError(true)
      setIsLoading(false)
      return { error: true }
    }
  }, [])

  const sharedState = {
    isLoading,
    subscription,
    error,
    subscribe: fetchSubscribe
  }

  return (
    <SubscriptionContext.Provider value={sharedState}>
      {children}
    </SubscriptionContext.Provider>
  )
}

export function useSubscription() {
  return useContext(SubscriptionContext)
}
