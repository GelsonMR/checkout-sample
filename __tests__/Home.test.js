import '@testing-library/jest-dom'
import { ThemeProvider } from 'styled-components'
import { render, screen } from '@testing-library/react'
import { OffersContext } from '../context/OffersContext'
import Home from '../pages/index'

const mockTheme = { colors: {} }

it('renders home', () => {
  const { container } = render(
    <ThemeProvider theme={mockTheme}>
      <Home />
    </ThemeProvider>
  )
  expect(container).toMatchSnapshot()
})

it('renders home with error', async () => {
  render(
    <OffersContext.Provider value={{ error: true }}>
      <ThemeProvider theme={mockTheme}>
        <Home />
      </ThemeProvider>
    </OffersContext.Provider>
  )

  expect(await screen.findByText('Erro ao carregar planos')).toBeInTheDocument()
})
