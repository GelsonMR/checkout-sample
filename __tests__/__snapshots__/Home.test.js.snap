// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home 1`] = `
<div>
  <header
    class="sc-2154031-0 ImzvS"
  >
    <a
      class="sc-2154031-1 fzbsAi"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Back"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </a>
    <div
      class="sc-2154031-2 bCwUhg"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="Whitebook logo"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </header>
  <div
    class="sc-98f575cc-0 DphZQ"
  >
    <div
      class="sc-e5c0085b-0 hatemx"
    >
      <h4
        class="sc-2764c71a-0 sc-eca3cee1-1 hiTdJi eWOeBD"
      >
        Estamos quase l√°!
      </h4>
      <span
        class="sc-2764c71a-1 sc-eca3cee1-2 gsTYYO eJMdPz"
      >
        Insira seus dados de pagamento abaixo:
      </span>
      <div
        class="sc-ab626f9a-0 kQztlJ"
      >
        <div
          class="sc-ab626f9a-1 eYZwYZ"
        >
          <div
            class="sc-ab626f9a-2 iDRWgr"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Master Card"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="sc-ab626f9a-2 iDRWgr"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Dinners Club"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="sc-ab626f9a-2 iDRWgr"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="American Express"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="sc-ab626f9a-2 iDRWgr"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="VISA"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="sc-ab626f9a-2 iDRWgr"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="ELO"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
        <span
          class="sc-2764c71a-1 sc-ab626f9a-3 gsTYYO fIHYrE"
        >
          Pagamentos por
           
          <div
            class="sc-ab626f9a-4 gPGzaw"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Iugu"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </span>
      </div>
      <form
        class="sc-eca3cee1-3 bGaSSX"
      >
        <div
          class="sc-7b786c8c-0 iILmjO"
        >
          <label
            class="sc-2764c71a-1 sc-2764c71a-3 sc-7b786c8c-2 gsTYYO dCDPex fFEuOT"
          >
            N√∫mero do cart√£o
            <input
              autocomplete="cc-number"
              class="sc-2764c71a-1 sc-7b786c8c-1 gsTYYO fuNbuP"
              name="creditCardNumber"
              placeholder="0000 0000 0000 0000"
            />
          </label>
        </div>
        <div
          class="sc-7b786c8c-0 iILmjO"
          style="grid-column: 1 / 2;"
        >
          <label
            class="sc-2764c71a-1 sc-2764c71a-3 sc-7b786c8c-2 gsTYYO dCDPex fFEuOT"
          >
            Validade
            <input
              autocomplete="cc-exp"
              class="sc-2764c71a-1 sc-7b786c8c-1 gsTYYO fuNbuP"
              name="creditCardExpirationDate"
              placeholder="MM/AA"
            />
          </label>
        </div>
        <div
          class="sc-7b786c8c-0 iILmjO"
          style="grid-column: 2 / 3;"
        >
          <label
            class="sc-2764c71a-1 sc-2764c71a-3 sc-7b786c8c-2 gsTYYO dCDPex fFEuOT"
          >
            CVV
            <input
              autocomplete="cc-csc"
              class="sc-2764c71a-1 sc-7b786c8c-1 gsTYYO fuNbuP"
              name="creditCardCVV"
              placeholder="000"
            />
          </label>
        </div>
        <div
          class="sc-7b786c8c-0 iILmjO"
        >
          <label
            class="sc-2764c71a-1 sc-2764c71a-3 sc-7b786c8c-2 gsTYYO dCDPex fFEuOT"
          >
            Nome impresso no cart√£o
            <input
              autocomplete="cc-name"
              class="sc-2764c71a-1 sc-7b786c8c-1 gsTYYO fuNbuP"
              name="creditCardHolder"
              placeholder="Seu nome"
            />
          </label>
        </div>
        <div
          class="sc-7b786c8c-0 iILmjO"
        >
          <label
            class="sc-2764c71a-1 sc-2764c71a-3 sc-7b786c8c-2 gsTYYO dCDPex fFEuOT"
          >
            CPF
            <input
              class="sc-2764c71a-1 sc-7b786c8c-1 gsTYYO fuNbuP"
              name="creditCardCPF"
              placeholder="000.000.000-00"
            />
          </label>
        </div>
        <button
          class="sc-2764c71a-1 sc-2764c71a-2 sc-f225acc-0 gsTYYO fSNzmq hYOqpk"
        >
          Finalizar pagamento
        </button>
      </form>
    </div>
    <div
      class="sc-e5c0085b-0 hatemx"
    >
      <h4
        class="sc-2764c71a-0 sc-7df95e90-0 hiTdJi bKfnkt"
      >
        Confira seu plano:
      </h4>
      <span
        class="sc-2764c71a-1 sc-2764c71a-3 sc-7df95e90-2 gsTYYO dCDPex kxgKuG"
      >
        fulano@cicrano.com.br
      </span>
      <ul />
      <div
        class="sc-2764c71a-1 sc-2764c71a-3 sc-7df95e90-9 gsTYYO dCDPex hrZCZB"
      >
        Sobre a cobran√ßa
        <div
          class="sc-7df95e90-10 gsKZbp"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Ajuda"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
